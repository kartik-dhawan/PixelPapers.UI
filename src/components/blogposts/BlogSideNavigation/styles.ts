export const styles = {
  blogsNavDivider: { borderColor: "#222", margin: "8px 0px", opacity: 0.5 },
  blogsSideNavWrapper: { top: 32, margin: "0rem 1rem 0rem 2rem" },
  blogsSideNavTitle: {
    fontWeight: 700,
    textTransform: "uppercase",
    fontSize: {
      xs: "18px",
      xl: "22px",
    },
  },
  blogNavList: {
    display: "flex",
    flexDirection: "column",
    gap: {
      xs: "8px",
      xl: "12px",
    },
  },
  blogNavListItem: {
    "& a": {
      fontWeight: 500,
      color: "#222",
      textDecoration: "none",
      width: "100%",
      height: "100%",
      padding: "8px 16px",
      fontSize: {
        xs: "16px",
        xl: "20px",
      },
    },
    padding: 0,
    borderRadius: "4px",
  },
}
