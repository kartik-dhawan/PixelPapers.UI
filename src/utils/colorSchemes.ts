export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
