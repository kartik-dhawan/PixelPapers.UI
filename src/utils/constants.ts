export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7
