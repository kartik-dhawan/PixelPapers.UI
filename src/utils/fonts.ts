import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
