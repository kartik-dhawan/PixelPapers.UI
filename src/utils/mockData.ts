export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/6TVtXnjJIZ812Bxm6VBqz5/5b89ce3dced278570a2a663453d69a65/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}
