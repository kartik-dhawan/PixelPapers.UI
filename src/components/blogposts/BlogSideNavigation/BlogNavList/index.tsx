import { TravelBlogContentfulObject } from "@/utils/interfaces"
import { lightenHexColor } from "@/utils/methods"
import { List, ListItem } from "@mui/material"
import Link from "next/link"
import { styles } from "../styles"

interface BlogNavListProps {
  blogs: TravelBlogContentfulObject[]
  themeColor?: string
}

const BlogNavList = ({ blogs, themeColor }: BlogNavListProps) => {
  return (
    <List sx={{ display: "flex", flexDirection: "column", gap: "6px" }}>
      {blogs.map((item) => {
        return (
          <ListItem
            key={item.sys.id}
            sx={{
              ...styles.blogNavListItem,
              backgroundColor: lightenHexColor(themeColor ?? "#222", 0.9),
              "&:hover": {
                backgroundColor: lightenHexColor(themeColor ?? "#222", 0.8),
              },
            }}
          >
            <Link href={`/travel/${item.fields.blogUrlSlug}`}>
              {item.fields.blogTitle}
            </Link>
          </ListItem>
        )
      })}
    </List>
  )
}

export default BlogNavList
