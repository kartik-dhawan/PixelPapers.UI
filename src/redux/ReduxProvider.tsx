"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
