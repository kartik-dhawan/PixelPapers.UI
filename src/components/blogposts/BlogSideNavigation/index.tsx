import { dmSans } from "@/utils/fonts"
import { TravelBlogContentfulObject } from "@/utils/interfaces"
import { Box, Divider, Stack } from "@mui/material"
import BlogNavList from "./BlogNavList"
import { styles } from "./styles"

interface BlogSideNavigationProps {
  blogType: string
  themeColor?: string
  blogs: TravelBlogContentfulObject[]
}

const BlogSideNavigation = ({
  blogType,
  themeColor,
  blogs,
}: BlogSideNavigationProps) => {
  return (
    <Stack
      divider={<Divider sx={styles.blogsNavDivider} />}
      sx={styles.blogsSideNavWrapper}
      className={dmSans.className}
    >
      <Box sx={styles.blogsSideNavTitle}>More on {blogType}</Box>
      <BlogNavList blogs={blogs} themeColor={themeColor} />
    </Stack>
  )
}

export default BlogSideNavigation
