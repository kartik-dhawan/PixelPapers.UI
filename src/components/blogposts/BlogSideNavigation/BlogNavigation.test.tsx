import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import BlogSideNavigation from "."
import { BLOG_DATA_MOCK } from "@/utils/mockData"

describe("Blog Side Navigation Component", () => {
  it("-- it should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <BlogSideNavigation
          blogType="Travel"
          blogs={[BLOG_DATA_MOCK, BLOG_DATA_MOCK]}
          themeColor="#222"
        />
      </ReduxProvider>,
    )

    expect(tree).toMatchSnapshot()
  })
})
