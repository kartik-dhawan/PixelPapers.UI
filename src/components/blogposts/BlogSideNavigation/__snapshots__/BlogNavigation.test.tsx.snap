// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Side Navigation Component -- it should match the snapshot 1`] = `
<div
  className="MuiStack-root className css-17y8gsm-MuiStack-root"
>
  <div
    className="MuiStack-root css-3y1137-MuiStack-root"
  >
    <div
      className="MuiBox-root css-a3225i"
    >
      More on 
      Travel
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1bxgi69-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Hide
    </button>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9rad00-MuiDivider-root"
  />
  <ul
    className="MuiList-root MuiList-padding css-172qiqt-MuiList-root"
  >
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style={
        {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-yld962-MuiListItem-root"
            disabled={false}
          >
            <a
              aria-label="Click here to go to another article named: The Labyrith of Jodhpur"
              href="/travel/labyrinth-of-jodhpur"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              The Labyrith of Jodhpur
            </a>
          </li>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style={
        {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-yld962-MuiListItem-root"
            disabled={false}
          >
            <a
              aria-label="Click here to go to another article named: The Labyrith of Jodhpur"
              href="/travel/labyrinth-of-jodhpur"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              The Labyrith of Jodhpur
            </a>
          </li>
        </div>
      </div>
    </div>
  </ul>
</div>
`;
